bin/sozo/src/commands/events.rs

```diff
@@ -55,63 +58,61 @@ impl EventsArgs {
     pub fn run(self, config: &Config) -> Result<()> {
         config.tokio_handle().block_on(async {
             let ws = scarb::ops::read_workspace(config.manifest_path(), config)?;
-            let profile_config = ws.load_profile_config()?;
-
             let (world_diff, provider, _) =
                 utils::get_world_diff_and_provider(self.starknet, self.world, &ws).await?;
-
             let provider = Arc::new(provider);
 
-            let from_block = if let Some(world_block) =
-                profile_config.env.as_ref().and_then(|e| e.world_block)
-            {
-                Some(BlockId::Number(world_block))
-            } else {
-                self.from_block.map(BlockId::Number)
-            };
-
-            let to_block = self.to_block.map(BlockId::Number);
-            let keys = self
-                .events
-                .map(|e| vec![e.iter().map(|event| starknet_keccak(event.as_bytes())).collect()]);
-
-            let event_filter = EventFilter {
-                from_block,
-                to_block,
-                address: Some(world_diff.world_info.address),
-                keys,
-            };
+            let latest_block = provider.block_number().await?;
+            let from_block = self.from_block.unwrap_or_else(|| latest_block.saturating_sub(1000));
```

Comment:
As edited in the issue, please consider using the `world_block` from the config, if it exists. 👍